src/heap.py

```
@@ -23,33 +25,42 @@ def push(self, val):
                 self.heap[parent] = self.heap[pos]
                 self.heap[pos] = tmp
                 pos = parent
-                if pos % 2 == 1:
+                if pos == 0:
+                    break
+                elif pos % 2 == 1:
                     parent = pos // 2
                 else:
                     parent = (pos // 2) - 1
 
     def pop(self):
```

You should raise an exception when pop on an empty list, not just return none.